### DISCUSSION TITLE
EditAction on table widget does not open modal

### DISCUSSION THREAD

**Discussion by robinheidrich** [filamentphp/filament#13122] [Help]
### Package

Widgets

### Package Version

v3.2.86

### How can we help you?

When I try to use `EditAction` as record action in a widget, I always get this error:

```
Failed to call component method. Public method [edit] not found on component
```

```php
public function table(Table $table): Table
{
    return $table
        ...
        ->recordAction(EditAction::class);
}
```

It should show a modal with the form, just like on the resource page, instead of going to the non-existent edit page.

Is there a way to make this work?

**Comment by leandrocfe** [comment]
EditAction in the table is a row action
https://filamentphp.com/docs/3.x/tables/actions#row-actions

**Reply by robinheidrich** [reply]
I see. But doesn't this also work with row actions? I don't see anything in the documentation that says otherwise.

**Comment by dave-mills** [comment]
A TableWidget class doesn't have the same integrations as a table on a ListPage that's part of a Resource, so the pre-built CRUD actions won't work by default. 

It's like having a Filament Table on a custom livewire component, so I'm pretty sure you need to build that Edit action yourself to get it to work on the widget. (If anyone knows of a way to easily make the pre-built EditAction, DeleteAction etc to work, I'd love to know!)

**Reply by robinheidrich** [reply]
That's a shame, hope they add it one day.
